src/console_commands.cpp

```diff
@@ -36,15 +36,12 @@ ConsoleCommands::ConsoleCommands(p2pool* pool)
 	, m_loop{}
 	, m_shutdownAsync{}
 	, m_tty{}
+	, m_stdin_pipe{}
+	, m_stdin_handle()
 	, m_loopThread{}
 	, m_readBuf{}
 	, m_readBufInUse(false)
 {
-	if (uv_guess_handle(0) != UV_TTY) {
```

Comment:
Removing this check will result in `m_loop` and `m_shutdownAsync` leaking if there's no tty and no stdin pipe. This check should be instead extended to check for both options and fail if none available.

Reply:
So it's better to move the whole `LOGERR(1, "tty or named pipe is not available");` block here.